scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py

```diff
@@ -359,47 +424,63 @@ def test_verbosity():
 def test_tolerance_float32():
     """Check lobpcg for attainable tolerance in float32.
     """
-    rnd = np.random.RandomState(0)
+    rng = np.random.RandomState(0)
     n = 50
     m = 3
     vals = -np.arange(1, n + 1)
     A = diags([vals], [0], (n, n))
     A = A.astype(np.float32)
-    X = rnd.standard_normal((n, m))
+    X = rng.random((n, m))
     X = X.astype(np.float32)
     eigvals, _ = lobpcg(A, X, tol=1.25e-5, maxiter=50, verbosityLevel=0)
     assert_allclose(eigvals, -np.arange(1, 1 + m), atol=2e-5, rtol=1e-5)
 
 
 def test_random_initial_float32():
-    """Check lobpcg in float32 for specific initial.
+    """Check lobpcg in float32 runs in-place without warning.
```

Comment:
I still need this to do deal with the test failure locally:

```diff
--- a/scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py
+++ b/scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py
@@ -447,7 +447,9 @@ def test_random_initial_float32():
     A = A.astype(np.float32)
     X = rng.random((n, m))
     X = X.astype(np.float32)
-    eigvals, _ = lobpcg(A, X, tol=1e-3, maxiter=50, verbosityLevel=1)
+    with np.testing.suppress_warnings() as sup:
+        sup.filter(UserWarning)
+        eigvals, _ = lobpcg(A, X, tol=1e-3, maxiter=50, verbosityLevel=1)
     assert_allclose(eigvals, -np.arange(1, 1 + m), atol=1e-2)
```

Can you summarize the advantages of this PR over what we currently have for non-experts like me? We're aiming to branch for the next release in ~1 week, so trying to decide what will be ready to go in.

Reply:
@tylerjereddy - done, thanks!

This PR is a code/algorithm optimalization, leading to a noticeable 10-30% improvement that grows with the matrix size, see https://github.com/scipy/scipy/pull/17675#issuecomment-1458577687

Reply:
That's fine. Primarily we want to test the absence of warnings. If we get reports of warnings we can figure out where the inplace work is going wrong and fixing it (or adding a test for the warning being raised in this case)